VLC error

I got this error by building VLC today. I have two PC´s. One PC shows me the following error. The other produced a green screen and an restart of the PC.

10:22:40 ┌ vlc git  ..................................... [Updates found]
10:22:40 ├ Running uninstall...
10:22:42 ├ Running autoreconf...
10:23:54 ├ Running configure...
10:30:15 ├ Running install...
Likely error (tail of the failed operation logfile):
      |                  ^~~~~~~~~~~~~~~~~~
make[4]: *** [Makefile:27524: misc/medialibrary/libmedialibrary_plugin_la-medialib.lo] Error 1
make[4]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[3]: *** [Makefile:31837: install-recursive] Error 1
make[3]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[2]: *** [Makefile:32268: install] Error 2
make[2]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[1]: *** [Makefile:1573: install-recursive] Error 1
make[1]: Leaving directory '/build/vlc-git/build-64bit'
make: *** [Makefile:2042: install] Error 2
install failed. Check C:/media-autobuild_suite-master/build/vlc-git/build-64bit/ab-suite.install.log
This is required for other packages, so this script will exit.
10:43:00   Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/iAlQ.zip

Issue may be fixed. I'm getting a different error now, but that's from the linker, so compiling should have been successful before.

```
undefined reference to `qt_static_plugin_QtQuickControls2Plugin()'
undefined reference to `qt_static_plugin_QtQuickTemplates2Plugin()'
```

^ this deserves another issue.

@j-mas - please check build\vlc-git\build-##bit\ab-suite.install.log because the relevant error does not appear in the tail in the whole suite's compile log.

@LigH-de 

No, I still have the same mistake. With running install it takes 20 min. 
Then comes the known error message or the green crash screen.

15:25:15 ┌ vlc git  .................................. [Recently updated]
15:25:17 ├ Running autoreconf...
15:26:38 ├ Running configure...
15:33:14 ├ Running install...
Likely error (tail of the failed operation logfile):
      |                  ^~~~~~~~~~~~~~~~~~
make[4]: *** [Makefile:27557: misc/medialibrary/libmedialibrary_plugin_la-medialib.lo] Error 1
make[4]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[3]: *** [Makefile:31870: install-recursive] Error 1
make[3]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[2]: *** [Makefile:32301: install] Error 2
make[2]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[1]: *** [Makefile:1573: install-recursive] Error 1
make[1]: Leaving directory '/build/vlc-git/build-64bit'
make: *** [Makefile:2042: install] Error 2
install failed.

I attached the ab-suite.install.log:

[ab-suite.install.log](https://github.com/m-ab-s/media-autobuild_suite/files/4355179/ab-suite.install.log)

Thank you for your help! 

Best regards
j-mas

14:31:18 ┌ vlc git  ..................................... [Updates found]
14:31:19 ├ Running uninstall...
vlc-mabs.patch
        Patch could not be applied with `git am`. Continuing without patching.
14:31:21 ├ Running autoreconf...
14:32:51 ├ Running configure...
14:41:45 ├ Running install...
Likely error (tail of the failed operation logfile):
      |                  ^~~~~~~~~~~~~~~~~~
make[4]: *** [Makefile:27815: misc/medialibrary/libmedialibrary_plugin_la-medialib.lo] Error 1
make[4]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[3]: *** [Makefile:32128: install-recursive] Error 1
make[3]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[2]: *** [Makefile:32566: install] Error 2
make[2]: Leaving directory '/build/vlc-git/build-64bit/modules'
make[1]: *** [Makefile:1575: install-recursive] Error 1
make[1]: Leaving directory '/build/vlc-git/build-64bit'
make: *** [Makefile:2044: install] Error 2
install failed.

I seem to be making progress, except that now I'm stuck at this error:

```
In file included from H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/x86_64-w64-mingw32/bits/c++allocator.h:33,
                 from H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/bits/allocator.h:46,
                 from H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/memory:63,
                 from ../../modules/misc/medialibrary/fs/fs.h:24,
                 from ../../modules/misc/medialibrary/fs/fs.cpp:25:
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/ext/new_allocator.h: In instantiation of 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = vlc::medialibrary::SDDevice; _Args = {char*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; _Tp = vlc::medialibrary::SDDevice]':
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/bits/alloc_traits.h:482:2:   required from 'static void std::allocator_traits<std::allocator<_Tp1> >::construct(std::allocator_traits<std::allocator<_Tp1> >::allocator_type&, _Up*, _Args&& ...) [with _Up = vlc::medialibrary::SDDevice; _Args = {char*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; _Tp = vlc::medialibrary::SDDevice; std::allocator_traits<std::allocator<_Tp1> >::allocator_type = std::allocator<vlc::medialibrary::SDDevice>]'
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/bits/shared_ptr_base.h:548:39:   required from 'std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {char*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; _Tp = vlc::medialibrary::SDDevice; _Alloc = std::allocator<vlc::medialibrary::SDDevice>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]'
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/bits/shared_ptr_base.h:679:16:   required from 'std::__shared_count<_Lp>::__shared_count(_Tp*&, std::_Sp_alloc_shared_tag<_Alloc>, _Args&& ...) [with _Tp = vlc::medialibrary::SDDevice; _Alloc = std::allocator<vlc::medialibrary::SDDevice>; _Args = {char*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]'
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/bits/shared_ptr_base.h:1344:71:   required from 'std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<vlc::medialibrary::SDDevice>; _Args = {char*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; _Tp = vlc::medialibrary::SDDevice; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]'
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/bits/shared_ptr.h:359:59:   required from 'std::shared_ptr<_Tp>::shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<vlc::medialibrary::SDDevice>; _Args = {char*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}; _Tp = vlc::medialibrary::SDDevice]'
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/bits/shared_ptr.h:701:14:   required from 'std::shared_ptr<_Tp> std::allocate_shared(const _Alloc&, _Args&& ...) [with _Tp = vlc::medialibrary::SDDevice; _Alloc = std::allocator<vlc::medialibrary::SDDevice>; _Args = {char*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]'
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/bits/shared_ptr.h:717:39:   required from 'std::shared_ptr<_Tp> std::make_shared(_Args&& ...) [with _Tp = vlc::medialibrary::SDDevice; _Args = {char*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}]'
../../modules/misc/medialibrary/fs/fs.cpp:226:61:   required from here
H:/repos/media-autobuild_suite/msys64/mingw64/include/c++/9.3.0/ext/new_allocator.h:145:20: error: invalid new-expression of abstract class type 'vlc::medialibrary::SDDevice'
  145 |  noexcept(noexcept(::new((void *)__p)
      |                    ^~~~~~~~~~~~~~~~~~
  146 |        _Up(std::forward<_Args>(__args)...)))
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../modules/misc/medialibrary/fs/fs.cpp:32:
../../modules/misc/medialibrary/fs/device.h:32:7: note:   because the following virtual functions are pure within 'vlc::medialibrary::SDDevice':
   32 | class SDDevice : public IDevice
      |       ^~~~~~~~
In file included from ../../modules/misc/medialibrary/fs/fs.cpp:30:
H:/repos/media-autobuild_suite/local64/include/medialibrary/filesystem/IDevice.h:45:32: note: 	'virtual const string& medialibrary::fs::IDevice::scheme() const'
   45 |     virtual const std::string& scheme() const = 0;
      |                                ^~~~~~
H:/repos/media-autobuild_suite/local64/include/medialibrary/filesystem/IDevice.h:48:18: note: 	'virtual bool medialibrary::fs::IDevice::isNetwork() const'
   48 |     virtual bool isNetwork() const = 0;
      |                  ^~~~~~~~~
```

@1480c1, any thoughts?

I have the following gists:

* https://gist.github.com/moisespr123/2369978dc603ed1e67bdf7aba7304416
* https://gist.github.com/moisespr123/70e16f70f5d016c3d0f0dacbf97526f0

They get applied, but gives me the above issue. Seems to be something on gcc?

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -2590,7 +2590,8 @@ if [[ $vlc == y ]]; then
         # msys2's patches
         # Issues due to conflicting `vlc_module_name` between libvlc and libvlccore when linking vlc-static.exe and undefines.
         # having gpg-error after GCRYPT_LIBS causes some issues, and since it's already included in GCRYPT_LIBS
-        do_patch "https://gist.githubusercontent.com/1480c1/8c50a0867aa1afceac064d2162120dde/raw/vlc-mabs.patch" am
+        do_patch "https://gist.githubusercontent.com/moisespr123/2369978dc603ed1e67bdf7aba7304416/raw/f83d6cd9171765ee87140bfc1555529aa029f240/vlc-corrected-patch" am
+        do_patch "https://gist.githubusercontent.com/moisespr123/70e16f70f5d016c3d0f0dacbf97526f0/raw/ad70ecd5db0b6349cb47c85932ce5e408cbb7004/medialib-patch"
 
         do_autoreconf
         # All of the disabled are because of multiple issues both on the installed libs and on vlc's side.

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -2590,8 +2590,8 @@ if [[ $vlc == y ]]; then
         # msys2's patches
         # Issues due to conflicting `vlc_module_name` between libvlc and libvlccore when linking vlc-static.exe and undefines.
         # having gpg-error after GCRYPT_LIBS causes some issues, and since it's already included in GCRYPT_LIBS
-        do_patch "https://gist.githubusercontent.com/moisespr123/2369978dc603ed1e67bdf7aba7304416/raw/f83d6cd9171765ee87140bfc1555529aa029f240/vlc-corrected-patch" am
-        do_patch "https://gist.githubusercontent.com/moisespr123/70e16f70f5d016c3d0f0dacbf97526f0/raw/ad70ecd5db0b6349cb47c85932ce5e408cbb7004/medialib-patch"
+        do_patch "https://gist.githubusercontent.com/moisespr123/2369978dc603ed1e67bdf7aba7304416/raw/vlc-corrected-patch" am
+        do_patch "https://gist.githubusercontent.com/moisespr123/70e16f70f5d016c3d0f0dacbf97526f0/raw/medialib-patch"
 
         do_autoreconf
         # All of the disabled are because of multiple issues both on the installed libs and on vlc's side.

